src/neuroglancer/segmentation_display_state/frontend.ts

```diff
@@ -414,6 +420,20 @@ function makeRegisterSegmentWidgetEventHandlers(displayState: SegmentationDispla
     stickyChildren[template.visibleIndex].addEventListener('click', visibleCheckboxHandler);
     children[template.filterIndex].addEventListener('click', filterHandler);
     element.addEventListener('action:select-position', selectHandler);
```

Comment:
For efficiency (because segment widgets are created and updated very frequently in the segment list) these segment widgets are handled in a special way --- where we have a template that we clone and then add the required event handlers (as minimally as possible).

I think it would probably be best to avoid creating a ColorWidget at all (but still obtain the template from it), and just implement the necessary event handlers separately.